OpenRA.Mods.Common/Traits/Turreted.cs

```diff
@@ -125,11 +125,13 @@ void MoveTurret()
 			TurretFacing = Util.TickFacing(TurretFacing, df, info.TurnSpeed);
 		}
 
-		public bool FaceTarget(Actor self, Target target)
+		public bool FaceTarget(Actor self, Target target, AttackTurreted at)
 		{
 			if (self.IsDisabled())
 				return false;
 
+			attack = at;
```

Comment:
This caches the current active AttackTurreted trait for FaceTarget (instead of constantly looking up all available AttackTurreted traits which has an armament controlling this turret). I'm assuming that only one AttackTurreted trait will be active at a turret at a time.